lms/djangoapps/instructor/views/api.py

```diff
@@ -2707,10 +2707,141 @@ def start_certificate_regeneration(request, course_id):
+    :return: JsonResponse object with success/error message or certificate exception to sync on the frontend.
+    """
+    course_key = CourseKey.from_string(course_id)
+    # Validate request data and return error response in case of invalid data
+    try:
+        certificate_exception, student = parse_request_data(request)
+    except ValueError as error:
+        return JsonResponse({'success': False, 'message': error.message}, status=400)
+
+    # Add new Certificate Exception for the student passed in request data
+    if request.method == 'POST':
+        try:
+            exception = add_certificate_exception(course_key, student, certificate_exception)
+        except ValueError as error:
+            return JsonResponse({'success': False, 'message': error.message}, status=400)
+        return JsonResponse(exception)
+
+    # Remove Certificate Exception for the student passed in request data
+    elif request.method == 'DELETE':
+        try:
+            remove_certificate_exception(course_key, student)
+        except ValueError as error:
+            return JsonResponse({'success': False, 'message': error.message}, status=400)
+
+        return JsonResponse({})
+
+
+def add_certificate_exception(course_key, student, certificate_exception):
+    """
+    Add a certificate exception to CertificateWhitelist table. Raises ValueError in case of invalid data.
+
+    :param course_key: course key of the course for whom to add certificates exception
+    :param student: User object for whom to add Certificate Exception
+    :param certificate_exception: JSON representation of backbone CertificateException model
+    :return: CertificateWhitelist item in dict format containing all the keys required by
+             backbone model CertificateException.
+    """
+    if CertificateWhitelist.objects.filter(user=student, course_id=course_key, whitelist=True).count() > 0:
+        raise ValueError(
+            _("Student (username/email={user}) already in certificate exception list.").format(user=student.username)
+        )
+
+    certificate_white_list, __ = CertificateWhitelist.objects.get_or_create(
+        user=student,
+        course_id=course_key,
+        defaults={
+            'whitelist': True,
+            'notes': certificate_exception.get('notes', '')
+        }
+    )
+
+    exception = dict({
+        'id': certificate_white_list.id,
+        'user_email': student.email,
+        'user_name': student.username,
+        'user_id': student.id,
+        'created': certificate_white_list.created.strftime("%A, %B %d, %Y"),
+    })
+
+    return exception
+
+
+def remove_certificate_exception(course_key, student):
+    """
+    Remove certificate exception for given course and student from CertificateWhitelist table and
+    invalidate its GeneratedCertificate if present.
+
+    :param course_key: course key of the course for whom to delete certificate.
+    :param student: User object for whom to add Certificate Exception
```

Comment:
Docstring says `:param student: User object for whom to add Certificate Exception` I don't we are adding Certificate Exception in this method.